Source/Foundation/bsfUtility/Private/Unix/BsUnixFileSystem.cpp

```diff
@@ -336,12 +336,12 @@ namespace bs
 		}
 
 		tmpdir.append("/banshee-XXXXXX");
-		size_t len = tmpdir.size()+1;
-		char *nameTemplate = bs_newN<char>(len);
-		snprintf(nameTemplate, len, "%s", tmpdir.c_str());
+		Vector<char> nameTemplate(tmpdir.data(),tmpdir.data()+tmpdir.size());
+		nameTemplate.push_back(0); // null terminate the name template.
```

Comment:
You're doing two allocations here.

One for the vector itself, and a second when you call push_back().

Instead, create an empty vector (or string, since it's a vector of char, std::string would be more directly analogous), and reserve the total size you want to use. Then push_back().

Reply:
You're right, the size of nameTemplate will be exactly equal to tmpdir.size(), I'll replace it with an one-liner.